pkg/cluster/px/px.go

```diff
@@ -768,6 +786,74 @@ func (ops *pxClusterOps) parseKvdbFromDaemonset() ([]string, map[string]string,
 	return endpoints, opts, clusterName, nil
 }
 
+func (ops *pxClusterOps) deleteAllPXComponents() error {
+	logrus.Infof("deleting all PX Kubernetes components from the cluster")
+
+	dss, err := ops.getPXDaemonsets(pxInstallTypeOCI)
+	if err != nil {
+		return err
+	}
+
+	for _, ds := range dss {
+		err = ops.k8sOps.DeleteDaemonSet(ds.Name, ds.Namespace)
+		if err != nil {
+			return err
+		}
+	}
+
+	depNames := [2]string{storkControllerName, storkSchedulerName}
+	for _, depName := range depNames {
+		err = ops.k8sOps.DeleteDeployment(depName, pxDefaultNamespace)
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	clusterRoles := [3]string{pxClusterRoleName, storkControllerClusterRole, storkSchedulerClusterRole}
```

Comment:
should we just log the failure and continue deleting others? Basically do a best effort delete of all the resources. 
For ex. in kops when you delete a cluster, it keeps on trying to delete all the AWS resources and never returns the command line call until all the resources are deleted. We don't need to be that aggressive but we can simply log all the failed attempts and exit.

Reply:
Apart from the not found error, any other error is unexpected. So I want to consider such unexpected errors as fatal. Having it best-effort will make it difficult to debug and the end result unpredictable since the job always succeeds but it may or may not have encountered errors.

Reply:
If a job always succeeds then it is even more confusing because anyone would run this tool and expect that everything is cleaned up. And they will expect subsequent kubectl apply's create's to succeed which will not.